service.go

```diff
@@ -21,7 +23,9 @@ type Service interface {
 	ServiceName() string
 	TrackName() string
 	URLRegex(string) bool
+	SearchRegex(string) bool
```

Comment:
The boolean statement in URLRegex is a bit more complicated. I think that it would be cleaner to get rid of the extra SearchRegex function.